backend/src/packages/orders/order.service.ts

```diff
@@ -0,0 +1,249 @@
+import { type UserEntityObjectWithGroupT } from '../users/users.js';
+import { OrderStatus, UserGroupKey } from './libs/enums/enums.js';
+import {
+  type OrderCreateRequestDto,
+  type OrderCreateResponseDto,
+  type OrderEntity as OrderEntityT,
+  type OrderFindByIdResponseDto,
+  type OrderUpdateRequestDto,
+  type OrderUpdateResponseDto,
+} from './libs/types/types.js';
+import { OrderEntity } from './order.entity.js';
+import { type OrderRepository } from './order.repository.js';
+
+class OrderService implements Omit<IService, 'find'> {
+  private orderRepository: OrderRepository;
+
+  private businessService: BusinessService;
+
+  private driverService: DriverService;
+
+  public constructor({
+    businessService,
+    orderRepository,
+    driverService,
+  }: {
+    orderRepository: OrderRepository;
+    businessService: BusinessService;
+    driverService: DriverService;
+  }) {
+    this.orderRepository = orderRepository;
+
+    this.businessService = businessService;
+
+    this.driverService = driverService;
+  }
+
+  public async create(
+    payload: OrderCreateRequestDto & {
+      userId: number | null;
+    },
+  ): Promise<OrderCreateResponseDto> {
+    const {
+      scheduledTime,
+      carsQty,
+      startPoint,
+      endPoint,
+      customerName,
+      customerPhone,
+      driverId,
+      userId,
+    } = payload;
+    const driver = await this.driverService.findById(driverId);
+
+    if (!driver) {
+      throw new NotFoundError({ message: HttpMessage.DRIVER_DOES_NOT_EXIST });
+    }
+    const order = await this.orderRepository.create(
+      OrderEntity.initializeNew({
+        price: 100, //Mock, get price from truck and calculated distance
+        scheduledTime,
+        carsQty,
+        startPoint,
+        endPoint,
+        status: OrderStatus.PENDING,
+        userId,
+        businessId: driver.businessId,
+        driverId,
+        customerName,
+        customerPhone,
+      }),
+    );
+
+    return order.toObject();
+  }
+
+  public async findById(id: OrderEntityT['id']): Promise<OrderEntityT | null> {
+    const foundOrder = await this.orderRepository.findById(id);
+
+    if (!foundOrder) {
+      throw new NotFoundError({
+        message: HttpMessage.ORDER_DOES_NOT_EXIST,
+      });
+    }
+
+    return foundOrder.toObject();
+  }
+
+  public async update(parameters: {
+    id: OrderEntityT['id'];
+    payload: OrderUpdateRequestDto;
+  }): Promise<OrderUpdateResponseDto> {
+    const updatedOrder = await this.orderRepository.update(parameters);
+
+    if (!updatedOrder) {
+      throw new NotFoundError({
+        message: HttpMessage.ORDER_DOES_NOT_EXIST,
+      });
+    }
+
+    return updatedOrder.toObject();
+  }
+
+  public async findOne({
+    id,
+    user,
+  }: {
+    id: OrderEntityT['id'];
+    user: UserEntityObjectWithGroupT | null;
+    businessId?: number;
+  }): Promise<OrderFindByIdResponseDto | null> {
+    let businessId;
+
```

Comment:
you don't need to create let variable which is not initailized

Reply:
I need it because I set value inside one if statement and use it in another if statement, if I not initialize it in outer scope it will not be visible in scope of second if statement

Reply:
I left an example below how you can rewrite method in service